src/hotspot/share/memory/virtualspace.cpp

```diff
@@ -610,7 +612,7 @@ void ReservedHeapSpace::initialize_compressed_heap(const size_t size, size_t ali
 }
 
 ReservedHeapSpace::ReservedHeapSpace(size_t size, size_t alignment, size_t page_size, const char* heap_allocation_directory) : ReservedSpace() {
-
+  set_nmt_flag(mtJavaHeap);
```

Comment:
It seems odd that we only initialize the _nmt_flag when `size == 0`. Could this be done after that check? If not, why not?

There's also a call to record_virtual_memory_type further down in the code. Why is that needed? Why isn't it enough to pass in the correct type to the os::reserve_memory call in the initialize function?

Reply:
Corrected.